src/containers/report_container.js

```diff
@@ -1,3 +1,89 @@
-const ReportContainer = () => null
+const assign = require('object-assign')
+const React = require('react')
+const { connect } = require('react-redux')
+const { PropTypes } = React
 
-module.exports = ReportContainer
+const FeatureModal = require('../components/feature_modal')
+const MFPropTypes = require('../util/prop_types')
+const MapView = require('../components/map_view')
+const getFieldMapping = require('../selectors/field_mapping')
+const getFilteredFeatures = require('../selectors/filtered_features')
+const getMapGeoJSON = require('../selectors/map_geojson')
+
+const LABEL_CHARS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+
+const styles = {
+  report: {
+    display: 'flex',
+    flexGrow: 1,
+    flexWrap: 'wrap',
+    overflow: 'scroll'
+  },
+  mapView: {
+    height: '500px',
+    width: '100%'
+  }
+}
+
+class ReportContainer extends React.Component {
+  static propTypes = {
+    features: PropTypes.array.isRequired,
+    fieldMapping: MFPropTypes.fieldMapping,
+    geojson: PropTypes.shape({
+      type: PropTypes.oneOf(['FeatureCollection']).isRequired,
+      features: PropTypes.arrayOf(MFPropTypes.mapViewFeature).isRequired
+    }).isRequired
+  }
+
+  render () {
+    const { features, fieldMapping, geojson } = this.props
+
+    return (
+      <div className='report container' style={styles.report}>
+        <h2>{ features.length } Observations</h2>
+        <MapView
+          fieldMapping={fieldMapping}
+          geojson={geojson}
+          style={styles.mapView}
+          disableScrollToZoom
+          labelPoints
+        />
+        {
+          features.map((feature, id) => (
+            <FeatureModal
+              key={id}
+              id={feature.properties.id}
+              label={feature.properties.__mf_label}
+            />
+          ))
+        }
+      </div>
+    )
+  }
+}
+
+function mapStateToProps (state) {
+  const features = getFilteredFeatures(state)
+
+  // TODO if features have been filtered, all of the markers should be the same color
+  // (this should already work, but fieldMapping.color is an unexpected value, e.g. 'area')
+  if (features.length < LABEL_CHARS.length) {
+    features.forEach((f, i) => {
+      f.properties.__mf_label = LABEL_CHARS.charAt(i)
+    })
+  }
+
+  const geojson = getMapGeoJSON(assign({}, state, {
```

Comment:
This will be causing re-rendering. getMapGeoJSON is a memoized function. but the argument is different on every call, so geojson will be different, so redux, which does a shallow compare on shallow compare to stop re-rendering, will continually rerender.